### ISSUE TITLE
python modules for AIX 7.1 (numpy/scipy, etc.,)

### ISSUE THREAD

**Issue by sanandece** [numpy/numpy#12907]
I am trying to install the below python modules on AIX. Can some one be able to get me the whl files that can be used on AIX. I am running AIX 7.1 (7100-05-03-1846)

I already have python 2.7.15-3 and pip installed on my LPAR. can one of you help?!

-pandas
-numpy
-scipy
-pydobc
-sklearn
-matplotlib
-sqlalchemy-teradata
-tabulate
-teradatasql
-teradatasqlalchemy
-sqlalchemy
-keras
-pyspark
-xgboost
-lightgbm
-catboost
-pytorch
-eli5
-nltk
-spacy
-scrapy
-gensim
-tensorflow
-statsmodel
-searborn

**Comment by tylerjereddy** [comment]
I don't think you can find wheels for AIX? The non-NumPy packages there are probably out of scope for us to deal with. There's some IBM / AIX-related discussion in #12688, including a link to some open NumPy issues with potentially-helpful discussions and some IBM team members chiming in.

**Comment by sanandece** [comment]
One of my colleagues recommended that I do the below

```bash-4.3# pip install numpy-1.16.0.zip
Processing ./numpy-1.16.0.zip
Installing collected packages: numpy
Running setup.py install for numpy ... done
Successfully installed numpy-1.16.0
bash-4.3# python
Python 2.7.15 (default, Jan  2 2019, 18:23:13)
[GCC 6.3.0] on aix6
Type "help", "copyright", "credits" or "license" for more information.
>>> import numpy
>>> exit ()
$ pip list
Package             Version
------------------- -------
iniparse            0.4
numpy               1.16.0
pip                 10.0.1
pycurl              7.19.3
pysqlite            1.1.7
setuptools          0.9.8
urlgrabber          3.10.1
yum-metadata-parser 1.1.4
```

Is this not the correct way to do it?

**Comment by rgommers** [comment]
@sanandece installing from source is the way to go I think.

How did you install Python?

**Comment by sanandece** [comment]
@rgommers I installed it using the Python RPM which I downloaded from the IBM AIX ToolBox site

**Comment by rgommers** [comment]
Thanks @sanandece, good to know. There are no wheels for that and no plans to create official AIX wheels any time soon (that fundamentally doesn't work with PyPI, there is no standard). However we are actively looking to fix any issues for building from source for the Python provided via the IBM AIX ToolBox.

Since there is nothing specifically actionable in this issue, I will close it. Feel free to report any issues you run into when building from source though.

**Comment by santezza** [comment]
hi everybody, I need help, I have an LPAR with Aix 7.1 , installed yum and when I try to install numpy it gave me errors every time.
Can anyone help me with that?
errors:

 blas_opt_info:
    blas_mkl_info:
    customize UnixCCompiler
      libraries mkl_rt not found in ['/usr/bin/../../opt/freeware/lib', '/usr/local/lib', '/usr/lib64', '/usr/lib', '/usr/lib/']
      NOT AVAILABLE

    blis_info:
    customize UnixCCompiler
      libraries blis not found in ['/usr/bin/../../opt/freeware/lib', '/usr/local/lib', '/usr/lib64', '/usr/lib', '/usr/lib/']
      NOT AVAILABLE

    openblas_info:
    customize UnixCCompiler
    customize UnixCCompiler
      libraries openblas not found in ['/usr/bin/../../opt/freeware/lib', '/usr/local/lib', '/usr/lib64', '/usr/lib', '/usr/lib/']
      NOT AVAILABLE

    atlas_3_10_blas_threads_info:
    Setting PTATLAS=ATLAS
    customize UnixCCompiler
      libraries tatlas not found in ['/usr/bin/../../opt/freeware/lib', '/usr/local/lib', '/usr/lib64', '/usr/lib', '/usr/lib/']
      NOT AVAILABLE

    atlas_3_10_blas_info:
    customize UnixCCompiler
      libraries satlas not found in ['/usr/bin/../../opt/freeware/lib', '/usr/local/lib', '/usr/lib64', '/usr/lib', '/usr/lib/']
      NOT AVAILABLE

    atlas_blas_threads_info:
    Setting PTATLAS=ATLAS
    customize UnixCCompiler
      libraries ptf77blas,ptcblas,atlas not found in ['/usr/bin/../../opt/freeware/lib', '/usr/local/lib', '/usr/lib64', '/usr/lib', '/usr/lib/']
      NOT AVAILABLE

    atlas_blas_info:
    customize UnixCCompiler
      libraries f77blas,cblas,atlas not found in ['/usr/bin/../../opt/freeware/lib', '/usr/local/lib', '/usr/lib64', '/usr/lib', '/usr/lib/']
      NOT AVAILABLE

    accelerate_info:
      NOT AVAILABLE

    /tmp/pip-install-u7erfhov/numpy/numpy/distutils/system_info.py:625: UserWarning:
        Atlas (http://math-atlas.sourceforge.net/) libraries not found.
        Directories to search for the libraries can be specified in the
        numpy/distutils/site.cfg file (section [atlas]) or by setting
        the ATLAS environment variable.
      self.calc_info()
    blas_info:
    customize UnixCCompiler
    customize UnixCCompiler
    C compiler: gcc -maix64 -fPIC -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -O2 -I/usr/include -I/opt/freeware/include -I/opt/freeware/include/ncurses -DAIX_GENUINE_CPLUSCPLUS -D_LINUX_SOURCE_COMPAT -Wl,-brtl -O2 -fPIC -I/usr/include -I/opt/freeware/include -I/opt/freeware/include/ncurses -DAIX_GENUINE_CPLUSCPLUS -D_LINUX_SOURCE_COMPAT -Wl,-brtl -O2 -fPIC

    creating /tmp/tmpqf6bh86m/tmp
    creating /tmp/tmpqf6bh86m/tmp/tmpqf6bh86m
    compile options: '-I/usr/include -I/usr/bin/../../opt/freeware/include -c'
    gcc: /tmp/tmpqf6bh86m/source.c
    Could not load program /opt/freeware/libexec/gcc/powerpc-ibm-aix7.1.0.0/6.3.0/cc1:
    Symbol resolution failed for cc1 because:
        Symbol _GLOBAL__AIXI_libgmp_so (number 169) is not exported from dependent
          module /usr/lib/libgmp.a(libgmp.so.10).
        Symbol _GLOBAL__AIXD_libgmp_so (number 170) is not exported from dependent
          module /usr/lib/libgmp.a(libgmp.so.10).
    Examine .loader section symbols with the 'dump -Tv' command.
    Could not load program /opt/freeware/libexec/gcc/powerpc-ibm-aix7.1.0.0/6.3.0/cc1:
    Symbol resolution failed for cc1 because:
        Symbol _GLOBAL__AIXI_libgmp_so (number 169) is not exported from dependent
          module /usr/lib/libgmp.a(libgmp.so.10).
        Symbol _GLOBAL__AIXD_libgmp_so (number 170) is not exported from dependent
          module /usr/lib/libgmp.a(libgmp.so.10).
    Examine .loader section symbols with the 'dump -Tv' command.
    customize UnixCCompiler
      FOUND:
        libraries = ['blas', 'blas']
        library_dirs = ['/usr/bin/../../opt/freeware/lib']
        include_dirs = ['/usr/include', '/usr/bin/../../opt/freeware/include']

      FOUND:
        define_macros = [('NO_ATLAS_INFO', 1)]
        libraries = ['blas', 'blas']
        library_dirs = ['/usr/bin/../../opt/freeware/lib']
        include_dirs = ['/usr/include', '/usr/bin/../../opt/freeware/include']

    /bin/sh: svnversion:  not found
    non-existing path in 'numpy/distutils': 'site.cfg'
    /bin/sh: svnversion:  not found
    F2PY Version 2
    lapack_opt_info:
    lapack_mkl_info:
    customize UnixCCompiler
      libraries mkl_rt not found in ['/usr/bin/../../opt/freeware/lib', '/usr/local/lib', '/usr/lib64', '/usr/lib', '/usr/lib/']
      NOT AVAILABLE

    openblas_lapack_info:
    customize UnixCCompiler
    customize UnixCCompiler
      libraries openblas not found in ['/usr/bin/../../opt/freeware/lib', '/usr/local/lib', '/usr/lib64', '/usr/lib', '/usr/lib/']
      NOT AVAILABLE

    openblas_clapack_info:
    customize UnixCCompiler
    customize UnixCCompiler
      libraries openblas,lapack not found in ['/usr/bin/../../opt/freeware/lib', '/usr/local/lib', '/usr/lib64', '/usr/lib', '/usr/lib/']
      NOT AVAILABLE

    atlas_3_10_threads_info:
    Setting PTATLAS=ATLAS
    customize UnixCCompiler
      libraries tatlas,tatlas not found in /usr/bin/../../opt/freeware/lib
    customize UnixCCompiler
      libraries lapack_atlas not found in /usr/bin/../../opt/freeware/lib
    customize UnixCCompiler
      libraries tatlas,tatlas not found in /usr/local/lib
    customize UnixCCompiler
      libraries lapack_atlas not found in /usr/local/lib
    customize UnixCCompiler
      libraries tatlas,tatlas not found in /usr/lib64
    customize UnixCCompiler
      libraries lapack_atlas not found in /usr/lib64
    customize UnixCCompiler
      libraries tatlas,tatlas not found in /usr/lib
    customize UnixCCompiler
      libraries lapack_atlas not found in /usr/lib
    customize UnixCCompiler
      libraries tatlas,tatlas not found in /usr/lib/
    customize UnixCCompiler
      libraries lapack_atlas not found in /usr/lib/
    <class 'numpy.distutils.system_info.atlas_3_10_threads_info'>
      NOT AVAILABLE

    atlas_3_10_info:
    customize UnixCCompiler
      libraries satlas,satlas not found in /usr/bin/../../opt/freeware/lib
    customize UnixCCompiler
      libraries lapack_atlas not found in /usr/bin/../../opt/freeware/lib
    customize UnixCCompiler
      libraries satlas,satlas not found in /usr/local/lib
    customize UnixCCompiler
      libraries lapack_atlas not found in /usr/local/lib
    customize UnixCCompiler
      libraries satlas,satlas not found in /usr/lib64
    customize UnixCCompiler
      libraries lapack_atlas not found in /usr/lib64
    customize UnixCCompiler
      libraries satlas,satlas not found in /usr/lib
    customize UnixCCompiler
      libraries lapack_atlas not found in /usr/lib
    customize UnixCCompiler
      libraries satlas,satlas not found in /usr/lib/
    customize UnixCCompiler
      libraries lapack_atlas not found in /usr/lib/
    <class 'numpy.distutils.system_info.atlas_3_10_info'>
      NOT AVAILABLE

**Comment by incgnito** [comment]
Downloaded the numpy source pkg from python website. Got below error while performing installation.



numpy/random/mtrand/mtrand.c:45210:17: error: 'PyThreadState {aka struct _ts}' has no member named 'exc_traceback'; did you mean 'curexc_trac
eback'?
     *tb = tstate->exc_traceback;
                 ^~
numpy/random/mtrand/mtrand.c: In function '__Pyx__ExceptionReset':
numpy/random/mtrand/mtrand.c:45217:22: error: 'PyThreadState {aka struct _ts}' has no member named 'exc_type'; did you mean 'curexc_type'?
     tmp_type = tstate->exc_type;
                      ^~
numpy/random/mtrand/mtrand.c:45218:23: error: 'PyThreadState {aka struct _ts}' has no member named 'exc_value'; did you mean 'curexc_value'?
     tmp_value = tstate->exc_value;
                       ^~
numpy/random/mtrand/mtrand.c:45219:20: error: 'PyThreadState {aka struct _ts}' has no member named 'exc_traceback'; did you mean 'curexc_trac
eback'?
     tmp_tb = tstate->exc_traceback;
                    ^~
numpy/random/mtrand/mtrand.c:45220:11: error: 'PyThreadState {aka struct _ts}' has no member named 'exc_type'; did you mean 'curexc_type'?
     tstate->exc_type = type;
           ^~
numpy/random/mtrand/mtrand.c:45221:11: error: 'PyThreadState {aka struct _ts}' has no member named 'exc_value'; did you mean 'curexc_value'?
     tstate->exc_value = value;
           ^~
numpy/random/mtrand/mtrand.c:45222:11: error: 'PyThreadState {aka struct _ts}' has no member named 'exc_traceback'; did you mean 'curexc_trac
eback'?
     tstate->exc_traceback = tb;
           ^~
numpy/random/mtrand/mtrand.c: In function '__Pyx__GetException':
numpy/random/mtrand/mtrand.c:45277:22: error: 'PyThreadState {aka struct _ts}' has no member named 'exc_type'; did you mean 'curexc_type'?
     tmp_type = tstate->exc_type;
                      ^~
numpy/random/mtrand/mtrand.c:45278:23: error: 'PyThreadState {aka struct _ts}' has no member named 'exc_value'; did you mean 'curexc_value'?
     tmp_value = tstate->exc_value;
                       ^~
numpy/random/mtrand/mtrand.c:45279:20: error: 'PyThreadState {aka struct _ts}' has no member named 'exc_traceback'; did you mean 'curexc_trac
eback'?
     tmp_tb = tstate->exc_traceback;
                    ^~
numpy/random/mtrand/mtrand.c:45280:11: error: 'PyThreadState {aka struct _ts}' has no member named 'exc_type'; did you mean 'curexc_type'?
     tstate->exc_type = local_type;
           ^~
numpy/random/mtrand/mtrand.c:45281:11: error: 'PyThreadState {aka struct _ts}' has no member named 'exc_value'; did you mean 'curexc_value'?
     tstate->exc_value = local_value;
           ^~
numpy/random/mtrand/mtrand.c:45282:11: error: 'PyThreadState {aka struct _ts}' has no member named 'exc_traceback'; did you mean 'curexc_trac
eback'?
     tstate->exc_traceback = local_tb;
           ^~
error: Command "gcc -maix64 -fPIC -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -O2 -I/usr/include -I/opt/freeware/include -I/opt/freew
are/include/ncurses -DAIX_GENUINE_CPLUSCPLUS -D_LINUX_SOURCE_COMPAT -Wl,-brtl -O2 -fPIC -I/usr/include -I/opt/freeware/include -I/opt/freewar
e/include/ncurses -DAIX_GENUINE_CPLUSCPLUS -D_LINUX_SOURCE_COMPAT -Wl,-brtl -O2 -fPIC -D_LARGE_FILES -Inumpy/core/include -Ibuild/src.aix-7.1
-3.7/numpy/core/include/numpy -Inumpy/core/src/private -Inumpy/core/src -Inumpy/core -Inumpy/core/src/npymath -Inumpy/core/src/multiarray -In
umpy/core/src/umath -Inumpy/core/src/npysort -I/opt/freeware/include/python3.7m -Ibuild/src.aix-7.1-3.7/numpy/core/src/private -Ibuild/src.ai
x-7.1-3.7/numpy/core/src/npymath -Ibuild/src.aix-7.1-3.7/numpy/core/src/private -Ibuild/src.aix-7.1-3.7/numpy/core/src/npymath -Ibuild/src.ai
x-7.1-3.7/numpy/core/src/private -Ibuild/src.aix-7.1-3.7/numpy/core/src/npymath -c numpy/random/mtrand/mtrand.c -o build/temp.aix-7.1-3.7/num
py/random/mtrand/mtrand.o -MMD -MF build/temp.aix-7.1-3.7/numpy/random/mtrand/mtrand.o.d" failed with exit status 1

**Comment by mattip** [comment]
@incgnito please open a new issue. Where did you get your python and what version is it?
